inference-engine/tools/vpu/vpu_graph_editor/main.cpp

```diff
@@ -0,0 +1,705 @@
+    size_t width() const { return _width; }
+    size_t height() const { return _height; }
+
+    std::shared_ptr<unsigned char> getData() {
+        return _data;
+    }
+};
+
+static short f32tof16(float x) {
+    static float min16 = asfloat((127 - 14) << 23);
+
+    static float max16 = asfloat(((127 + 15) << 23) | 0x007FE000);
+    static uint32_t max16f16 = ((15 + 15) << 10) | 0x3FF;
+
+    union {
+        float f;
+        uint32_t u;
+    } v{};
+    v.f = x;
+
+    uint32_t s = (v.u >> 16) & 0x8000;
+
+    v.u &= 0x7FFFFFFF;
+
+    if ((v.u & EXP_MASK_F32) == EXP_MASK_F32) {
+        if (v.u & 0x007FFFFF) {
+            return s | (v.u >> (23 - 10)) | 0x0200;
+        } else {
+            return s | (v.u >> (23 - 10));
+        }
+    }
+
+    float halfULP = asfloat(v.u & EXP_MASK_F32) * asfloat((127 - 11) << 23);
+    v.f += halfULP;
+
+    if (v.f < min16 * 0.5F) {
+        return s;
+    }
+
+    if (v.f < min16) {
+        return s | (1 << 10);
+    }
+
+    if (v.f >= max16) {
+        return max16f16 | s;
+    }
+
+    v.u -= ((127 - 15) << 23);
+
+    v.u >>= (23 - 10);
+
+    return v.u | s;
+}
+
+std::vector<std::uint8_t> loadImage(const std::string &imageFilename,std::shared_ptr<ngraph::opset3::Parameter>& param) {
+    auto& tens = param->output(0).get_tensor();
+    //auto layout = tens.get_tensor_layout();
+
+    BitMap reader(imageFilename);
+
+    const auto dims = tens.get_shape();
+
+    const size_t N = dims[0];
+    const size_t C = dims[1];
+    const size_t H = dims[2];
+    const size_t W = dims[3];
+
+    const size_t img_w = reader.width();
+    const size_t img_h = reader.height();
+
+    const size_t numImageChannels = reader.size() / (reader.width() * reader.height());
+    if (C != numImageChannels && C != 1) {
+        std::cout << "loadImage error: Input channels mismatch: image channels " << numImageChannels << ", "
+                  << "network channels " << C << ", expecting count of image channels are equal "
+                  << "to count if network channels or count of network channels are equal to 1" << std::endl;
+        std::exit(EXIT_FAILURE);
+    }
+
+    std::vector<std::uint8_t> info; 
+    const unsigned char* RGB8 = reader.getData().get();
+    const float xScale = 1.0f * img_w / W;
+    const float yScale = 1.0f * img_h / H;
+
+    for (int n = 0; n < N; n++) {
+        for (int h = 0; h < H; ++h) {
+            int y = static_cast<int>(std::floor((h + 0.5f) * yScale));
+            for (int w = 0; w < W; ++w) {
+                int x = static_cast<int>(std::floor((w + 0.5f) * xScale));
+                for (int c = 0; c < C; c++) {
+                    float tmp = 0.f;
+                    tmp = 1.0 * RGB8[(y * img_w + x) * numImageChannels + c];
+                    auto tmpVec = reinterpret_cast<std::uint8_t*>(&tmp);
+                    for(int i = 0; i < sizeof(float); ++i ) {
+                        info.push_back(tmpVec[i]);
```

Comment:
You can use `data()` method in `std::vector` and cast it to required type (e.g. float*).
And then just assign `1.0 * RGB8[(y * img_w + x) * numImageChannels + c];` to each element of raw data